contrib/opencensus-ext-azure/examples/custom.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from flask import Flask
+from opencensus.ext.flask.flask_middleware import FlaskMiddleware
+
+app = Flask(__name__)
+# TODO:
+# Replace 00000000-0000-0000-0000-000000000000 with your Instrumentation Key
+# https://docs.microsoft.com/en-us/azure/azure-monitor/app/create-new-resource
+app.config['OPENCENSUS_TRACE'] = {
```

Comment:
@c24t @songy23 I wish to know what's the long term configuration structure in your mind.

For example, today we put "PROPAGATOR" under "OPENCENSUS_TRACE", which means if we have tags propagation (e.g. we could have W3C correlation context protocol for tags propagation under HTTP, or a binary protocol for gRPC), either we need to introduce something like `OPENCENSUS_TAGS.PROPAGATOR`, or decouple propagator from traces (e.g. having a top level propagator concept, which takes the entire context, pick whatever is needed, and serialize/deserialize).
Similar question for log sampler, potentially we will need to have `OPENCENSUS_LOG.SAMPLER`.

Reply:
Chatted with @songy23 offline, it seems gRPC integration is taking the approach of separated propagators.

So, the configuration might look like:

```
app.config['OPENCENSUS'] = {
    'TRACE': {
        'SAMPLER': 'opencensus.trace.samplers.ProbabilitySampler(rate=1)',
        'EXPORTER': 'opencensus.ext.stackdriver.trace_exporter.StackdriverExporter()',
        'PROPAGATOR': 'google_cloud_format.GoogleCloudFormatPropagator()'
    },
    'METRICS':  {
        'EXPORTER': 'opencensus.ext.stackdriver.metrics_exporter.StackdriverExporter()'
    },
    'LOGS': {
        'SAMPLER': 'opencensus.trace.samplers.ProbabilitySampler(rate=1)',
        'EXPORTER': 'opencensus.ext.stackdriver.log_exporter.StackdriverExporter()'
    },
    'TAGS': {
        'PROPAGATOR': 'CorrelationContextPropagator()'
    }
}
```

@songy23 please correct me if I'm wrong.

Reply:
@c24t this also brings the question of where should samplers go, do we want to have ProbabilitySampler under `trace` or put it in the general place so both traces and logs can share them. Or should log respect the sampling decision if there is a trace context (and what if the customer is using log feature only, without using any of the trace features)?